### ISSUE TITLE
[Accessibility]: User dropdown doesn't focus when clicked

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#31495]
### Problem description

1. Go to https://sourcegraph.com and sign in
2. Open the user dropdown by clicking on the icon with your mouse
3. Press the down arrow key.
4. Notice that you can't navigate properly within the dropdown

### Expected behavior

Expected:
- We should be able to navigate easily within the dropdown with the arrow keys.

See this example that uses the Menu component: https://storybook.sgdev.org/?path=/story/wildcard-menu--menu-example

### Additional details

It seems this behavior only happens occasionally, see below video. Maybe a bug is breaking focus somehow.

https://user-images.githubusercontent.com/9516420/154727487-02788864-8af4-4c9d-b2b1-5b98d6cb6538.mp4

closed by https://github.com/sourcegraph/sourcegraph/pull/31539

**Comment by gitstart-sourcegraph** [comment]
@umpox We found out the root problem is from supporting controlled states for `wildcard/Menu` component - which we use  to support `openByDefault` prop and support hover to open menu in `NavDropdown` (reverted).

As you can see in the `Menu` implementation, we are binding `isOpen` prop directly into `<Popover />` but  it isn't synced with `isExpanded` from `ReachMenu`.

So suggest to remove controlled states implementation in `Menu` component. WDYT?
